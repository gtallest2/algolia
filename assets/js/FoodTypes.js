import React from 'react'

import { CuisinePlaceholder } from './placeholders'

const { func, object, string } = React.PropTypes

const FoodTypes = React.createClass({
  propTypes: {
    searchResults: object,
    currentFoodType: string,
    handleFoodTypeClick: func
  },

  renderFoodTypes () {
    if (!Object.keys(this.props.searchResults).length) { return }
    const foodTypes = this.props.searchResults.getFacetValues('food_type')
    return foodTypes.map((type, i) => {
      return (
        <li key={i} onClick={this.handleFoodTypeClick.bind(null, type.name)} className={this.props.currentFoodType === type.name ? 'active' : ''} >
          <span className='food-type'>{type.name}</span>
          <span className='food-type-amount'>{type.count}</span>
        </li>
      )
    }, this)
  },

  handleFoodTypeClick (type) {
    this.props.handleFoodTypeClick(type)
  },

  render () {
    const foodTypes = (
      <ul className='cuisine'>
        {this.renderFoodTypes()}
      </ul>
     )
    return (
      <div className='foodtypes-filter'>
        <h4>Cuisine/Food Type</h4>
        {!Object.keys(this.props.searchResults).length /* || !this.props.searchResults.facets.length */ ? CuisinePlaceholder() : foodTypes}
      </div>
    )
  }
})

export default FoodTypes
