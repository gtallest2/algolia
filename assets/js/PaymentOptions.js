import React from 'react'

const { func, bool } = React.PropTypes

const PaymentOptions = React.createClass({
  propTypes: {
    mobileMenu: bool,
    handlePaymentToggle: func
  },

  handlePaymentToggle (cardType) {
    this.props.handlePaymentToggle(cardType)
  },

  render () {
    return (
      <div className='payments-filter'>
        <h4>Payment Options</h4>
        <ul className='payment-options'>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'AMEX')} type='checkbox' />
            <span className='card cc-ae' />&nbsp;
            <span className='amex-card-name'>{this.props.mobileMenu ? 'AMEX' : 'American Express'}</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'Discover')} type='checkbox' />
            <span className='card cc-discover' />&nbsp;
            <span className='discover-card-name'>Discover</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'MasterCard')} type='checkbox' />
            <span className='card cc-mc' />&nbsp;
            <span className='mastercard-card-name'>MasterCard</span>
          </li>
          <li className='payment-type'>
            <input onClick={this.handlePaymentToggle.bind(null, 'Visa')} type='checkbox' />
            <span className='card cc-visa' />&nbsp;
            <span className='visa-card-name'>Visa</span>
          </li>
        </ul>
      </div>
    )
  }
})

export default PaymentOptions
