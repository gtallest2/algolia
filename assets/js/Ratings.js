import React from 'react'

const { func, number } = React.PropTypes

const Ratings = React.createClass({
  propTypes: {
    currentRatingFilter: number,
    handleRatingFilter: func
  },
  handleRatingFilter (value) {
    this.props.handleRatingFilter(value)
  },
  render () {
    return (
      <div className='ratings-filter'>
        <h4>Rating</h4>
        <ul className='ratings'>
          <li onClick={this.handleRatingFilter.bind(null, 0)} className={this.props.currentRatingFilter === 0 ? 'active stars no-star' : 'stars no-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 1)} className={this.props.currentRatingFilter === 1 ? 'active stars one-star' : 'stars one-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 2)} className={this.props.currentRatingFilter === 2 ? 'active stars two-star' : 'stars two-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 3)} className={this.props.currentRatingFilter === 3 ? 'active stars three-star' : 'stars three-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 4)} className={this.props.currentRatingFilter === 4 ? 'active stars four-star' : 'stars four-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 5)} className={this.props.currentRatingFilter === 5 ? 'active stars five-star' : 'stars five-star'} />
        </ul>
      </div>
    )
  }
})

export default Ratings
