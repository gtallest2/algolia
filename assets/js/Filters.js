import React from 'react'
import onClickOutside from 'react-onclickoutside'

import FoodTypes from './FoodTypes'
import Ratings from './Ratings'
import PaymentOptions from './PaymentOptions'

const { bool, object, func } = React.PropTypes

const Filters = onClickOutside(React.createClass({
  propTypes: {
    toggleMobile: func,
    mobileMenu: bool,
    searchResults: object,
    foodFacet: func,
    addPayment: func,
    removePayment: func,
    addRating: func,
    removeRating: func,
    blurMenu: func,
    combineAddAndRemove: func
  },

  getInitialState () {
    return {
      currentRatingFilter: 10,
      currentFoodType: '',
      payments: {
        AMEX: false,
        Discover: false,
        MasterCard: false,
        Visa: false
      }
    }
  },

  handleFoodTypeClick (facetValue) {
    this.state.currentFoodType === ''
      ? this.setState({ currentFoodType: facetValue })
      : this.setState({ currentFoodType: '' })
    this.props.foodFacet('food_type', facetValue)
  },

  addPaymentDisjunctive (value) {
    const paymentState = Object.assign(this.state.payments, { [value]: true })
    this.setState({ payments: paymentState })
    this.props.addPayment('payment_options', value)
  },

  removePaymentDisjunctive (value) {
    const paymentState = Object.assign(this.state.payments, { [value]: false })
    this.setState({ payments: paymentState })
    this.props.removePayment('payment_options', value)
  },

  handlePaymentToggle (value) {
    this.state.payments[value] ? this.removePaymentDisjunctive(value) : this.addPaymentDisjunctive(value)
  },

  handleRatingFilter (value) {
    if (this.state.currentRatingFilter === 10) {
      this.props.addRating(value)
      this.setState({ currentRatingFilter: value })
    } else {
      if (this.state.currentRatingFilter === value) {
        this.props.removeRating(value)
        this.setState({ currentRatingFilter: 10 })
      } else {
        this.props.combineAddAndRemove(value, this.state.currentRatingFilter)
        this.setState({ currentRatingFilter: value })
      }
    }
  },

  closeMenu () {
    this.props.toggleMobile()
  },

  handleClickOutside () {
    this.props.blurMenu()
  },

  render () {
    return (
      <div style={this.props.mobileMenu ? { transform: 'translateX(0)' } : {}} className='filters' >
        <div className='mobile-header'>
          <div className='mobile-menu-title'>Sort by:</div>
          <div onClick={this.closeMenu} className='mobile-menu-close'>x</div>
        </div>
        <FoodTypes
          searchResults={this.props.searchResults}
          currentFoodType={this.state.currentFoodType}
          handleFoodTypeClick={this.handleFoodTypeClick}
         />
        <Ratings
          currentRatingFilter={this.state.currentRatingFilter}
          handleRatingFilter={this.handleRatingFilter}
        />
        <PaymentOptions
          mobileMenu={this.props.mobileMenu}
          handlePaymentToggle={this.handlePaymentToggle}
        />
      </div>
    )
  }
}))

export default Filters
