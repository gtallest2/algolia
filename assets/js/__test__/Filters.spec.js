import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Filters from '../Filters'
import FoodTypes from '../FoodTypes'
import Ratings from '../Ratings'
import PaymentOptions from '../PaymentOptions'

import exampleSearch from '../../../example-search'

test('Filters snapshot test', () => {
  const component = shallow(<Filters />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})


// test('Filters renders correct components', () => {
//   const component = shallow(
//     <Filters searchResults={exampleSearch}/>
//   )
//   console.log(component.html());
// })
