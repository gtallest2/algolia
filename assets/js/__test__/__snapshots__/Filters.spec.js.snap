exports[`test Filters snapshot test 1`] = `
<Filters
  disableOnClickOutside={[Function]}
  enableOnClickOutside={[Function]} />
`;
